Modules/audioop.c

```diff
@@ -1897,31 +1903,51 @@ static PyMethodDef audioop_methods[] = {
     { 0,          0 }
 };
 
+static int
+audioop_traverse(PyObject *m, visitproc visit, void *arg) {
+    Py_VISIT(_audioopstate(m)->AudioopError);
+    return 0;
+}
+static int
+audioop_clear(PyObject *m) {
+    Py_CLEAR(_audioopstate(m)->AudioopError);
+    return 0;
+}
+static void
+audioop_free(void *m) {
+    audioop_clear((PyObject *)m);
+}
 
 static struct PyModuleDef audioopmodule = {
     PyModuleDef_HEAD_INIT,
     "audioop",
     NULL,
-    -1,
+    sizeof(_audioopstate),
     audioop_methods,
     NULL,
-    NULL,
-    NULL,
-    NULL
+    audioop_traverse,
+    audioop_clear,
+    audioop_free
 };
 
 PyMODINIT_FUNC
 PyInit_audioop(void)
 {
-    PyObject *m, *d;
+    PyObject *m;
+    m = PyState_FindModule(&audioopmodule);
+    if (m != NULL) {
+        Py_INCREF(m);
+        return m;
+    }
     m = PyModule_Create(&audioopmodule);
     if (m == NULL)
         return NULL;
-    d = PyModule_GetDict(m);
-    if (d == NULL)
-        return NULL;
-    AudioopError = PyErr_NewException("audioop.error", NULL, NULL);
-    if (AudioopError != NULL)
-         PyDict_SetItemString(d,"error",AudioopError);
+    PyObject *AudioopError = PyErr_NewException("audioop.error", NULL, NULL);
+    if (AudioopError != NULL) {
```

Comment:
Let's follow what most extensions do and return `NULL` if this fails. An example of this is: [`_localemodule`](https://github.com/python/cpython/blob/f2f55e7f03d332fd43bc665a86d585a79c3b3ed4/Modules/_localemodule.c#L751-L756)